mongo/integration/client_side_encryption_prose_test.go

```diff
@@ -1001,3 +1115,100 @@ func decodeJSONFile(mt *mtest.T, file string, val interface{}) bson.Raw {
 func rawValueToCoreValue(rv bson.RawValue) bsoncore.Value {
 	return bsoncore.Value{Type: rv.Type, Data: rv.Value}
 }
+
+type deadlockTest struct {
+	clientTest           *mongo.Client
+	clientKeyVaultOpts   *options.ClientOptions
+	clientKeyVaultEvents []startedEvent
+	clientEncryption     *mongo.ClientEncryption
+	ciphertext           primitive.Binary
+}
+type startedEvent struct {
+	cmd string
+	db  string
+}
+
+func compareStarted(mt *mtest.T, clientName string, actual []startedEvent, expected []startedEvent) {
```

Comment:
If you change `startedEvent` to have exported `Command` and `DB` fields, you can replace this entire function with a single `assert.Equal` call. The `assert` library internally uses [go-cmp](https://pkg.go.dev/github.com/google/go-cmp/cmp) and the documentation for the [Equal](https://pkg.go.dev/github.com/google/go-cmp/cmp#Equal) function says this about comparing slices:

> Slices are equal if they are both nil or both non-nil, where recursively calling Equal on all non-ignored slice or array elements report equal.

I tried this locally. The tests all passed and modifying any of the expectations causes them to fail. Note that you have to change `startedEvent` fields to be exported because go-cmp doesn't compare unexported fields by default.

Reply:
Oh that is slick. Thanks for the background, and for trying this out locally!